### PR TITLE
HBASE-22081 master shutdown: close RpcServer and procWAL first thing

### PR DESCRIPTION
**Pull request by sershe-ms**

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteNamespaceProcedure.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 package org.apache.hadoop.hbase.master.procedure;
 
 import java.io.FileNotFoundException;
-import java.io.IOException;
+import java.io.IOException;import java.io.InterruptedIOException;
```
Comment (tommyzli) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteNamespaceProcedure.java]: add the import to a new line? You have this in a few files

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 280 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 261 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 74 | master passed |
| +1 | shadedjars | 252 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 221 | master passed |
| +1 | javadoc | 30 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 227 | the patch passed |
| +1 | compile | 48 | the patch passed |
| +1 | javac | 48 | the patch passed |
| -1 | checkstyle | 68 | hbase-server: The patch generated 1 new + 320 unchanged - 0 fixed = 321 total (was 320) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 247 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 467 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| -1 | findbugs | 52 | hbase-server in the patch failed. |
| +1 | javadoc | 30 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 19101 | hbase-server in the patch failed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 21503 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.tool.TestSecureLoadIncrementalHFiles |
|   | hadoop.hbase.tool.TestLoadIncrementalHFiles |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.client.replication.TestReplicationAdmin |
|   | hadoop.hbase.master.procedure.TestMasterFailoverWithProcedures |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.client.TestConnectionImplementation |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.replication.TestReplicationSmallTestsSync |
|   | hadoop.hbase.client.TestHbck |
|   | hadoop.hbase.client.replication.TestReplicationAdminWithClusters |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAsyncTableAdminApi |
|   | hadoop.hbase.replication.TestReplicationKillSlaveRSWithSeparateOldWALs |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/200 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux fada6a5033d2 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 9743b3c70d |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/artifact/out/diff-checkstyle-hbase-server.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/artifact/out/patch-findbugs-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/testReport/ |
| Max. process+thread count | 4986 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 48 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 225 | master passed |
| +1 | compile | 50 | master passed |
| +1 | checkstyle | 69 | master passed |
| +1 | shadedjars | 254 | branch has no errors when building our shaded downstream artifacts. |
| -1 | findbugs | 154 | hbase-server in master failed. |
| +1 | javadoc | 30 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 229 | the patch passed |
| +1 | compile | 49 | the patch passed |
| +1 | javac | 49 | the patch passed |
| -1 | checkstyle | 66 | hbase-server: The patch generated 1 new + 320 unchanged - 0 fixed = 321 total (was 320) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 248 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 465 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 220 | the patch passed |
| +1 | javadoc | 32 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 19075 | hbase-server in the patch failed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 21312 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.quotas.TestSpaceQuotas |
|   | hadoop.hbase.tool.TestSecureLoadIncrementalHFiles |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.master.locking.TestLockManager |
|   | hadoop.hbase.client.replication.TestReplicationAdminWithClusters |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestAsyncTableAdminApi |
|   | hadoop.hbase.replication.TestReplicationKillSlaveRSWithSeparateOldWALs |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/200 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 98dbfbf7ddd6 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 9743b3c70d |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/artifact/out/branch-findbugs-hbase-server.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/testReport/ |
| Max. process+thread count | 5259 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 22 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 238 | master passed |
| +1 | compile | 54 | master passed |
| +1 | checkstyle | 74 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 213 | master passed |
| +1 | javadoc | 32 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 239 | the patch passed |
| +1 | compile | 52 | the patch passed |
| +1 | javac | 52 | the patch passed |
| -1 | checkstyle | 71 | hbase-server: The patch generated 1 new + 320 unchanged - 0 fixed = 321 total (was 320) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 249 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 512 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 232 | the patch passed |
| +1 | javadoc | 31 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8328 | hbase-server in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 10712 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/200 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 7196cb67d8ce 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 70296a2e78 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/4/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/4/testReport/ |
| Max. process+thread count | 4295 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 54 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 341 | master passed |
| +1 | compile | 72 | master passed |
| +1 | checkstyle | 89 | master passed |
| +1 | shadedjars | 323 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 212 | master passed |
| +1 | javadoc | 42 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 303 | the patch passed |
| +1 | compile | 62 | the patch passed |
| +1 | javac | 62 | the patch passed |
| -1 | checkstyle | 81 | hbase-server: The patch generated 1 new + 320 unchanged - 0 fixed = 321 total (was 320) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 299 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 693 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 255 | the patch passed |
| +1 | javadoc | 45 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 9498 | hbase-server in the patch passed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 12487 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/200 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 5503f412a707 4.4.0-141-generic #167~14.04.1-Ubuntu SMP Mon Dec 10 13:20:24 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 70296a2e78 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/3/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/3/testReport/ |
| Max. process+thread count | 4698 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-200/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Any progress here? Conflict at least needs to be addressed? Thanks.

**Comment by Apache9** [comment]
Close since the PR is stale. Feel free to reopen.

**Pull request closed without merging**